aws/resource_aws_quicksight_user_test.go

```diff
@@ -0,0 +1,223 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/quicksight"
+)
+
+func TestAccAWSQuickSightUser_basic(t *testing.T) {
+	var user quicksight.User
+	resourceName := "aws_quicksight_user.default"
+	rName1 := acctest.RandomWithPrefix("tf-acc-test")
+	rName2 := acctest.RandomWithPrefix("tf-acc-test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckQuickSightUserDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSQuickSightUserConfig(rName1),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckQuickSightUserExists(resourceName, &user),
+					resource.TestCheckResourceAttr(resourceName, "user_name", rName1),
+					testAccCheckResourceAttrRegionalARN(resourceName, "arn", "quicksight", fmt.Sprintf("user/default/%s", rName1)),
+				),
+			},
+			{
+				Config: testAccAWSQuickSightUserConfig(rName2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckQuickSightUserExists(resourceName, &user),
+					resource.TestCheckResourceAttr(resourceName, "user_name", rName2),
+					testAccCheckResourceAttrRegionalARN(resourceName, "arn", "quicksight", fmt.Sprintf("user/default/%s", rName2)),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSQuickSightUser_withRealEmail(t *testing.T) {
```

Comment:
Nit: This test doesn't appear to be testing real emails, so maybe worth just noting which attribute we are testing instead

```suggestion
func TestAccAWSQuickSightUser_Email(t *testing.T) {
```

Reply:
I renamed this to `TestAccAWSQuickSightUser_withInvalidFormattedEmailStillWorks` which is more what I was going for.